subworkflows/local/fusioninspector_workflow.nf

```diff
@@ -17,43 +17,49 @@ workflow FUSIONINSPECTOR_WORKFLOW {
         ch_arriba_ref_cytobands
         ch_hgnc_ref
         ch_hgnc_date
+        ch_starfusion_ref
+        skip_vis
+        skip_vcf
 
     main:
         ch_versions = Channel.empty()
         ch_arriba_visualisation = Channel.empty()
-        index ="${params.starfusion_ref}"
 
         ch_fusion_list = ( params.tools_cutoff > 1 ? fusion_list_filtered : fusion_list )
-        .branch{
-            no_fusions: it[1].size() == 0
-            fusions: it[1].size() > 0
-        }
 
         if (params.whitelist)  {
-            ch_whitelist = ch_fusion_list.fusions.combine(Channel.value(file(params.whitelist, checkIfExists:true)))
+            ch_whitelist = ch_fusion_list.combine(Channel.value(file(params.whitelist, checkIfExists:true)))
                             .map { meta, fusions, whitelist -> [ meta, [fusions, whitelist] ] }
 
             CAT_CAT(ch_whitelist) // fusioninspector takes care of possible duplicates
             ch_versions = ch_versions.mix(CAT_CAT.out.versions)
             ch_reads_fusion = reads.join(CAT_CAT.out.file_out )
         }
         else {
-            ch_reads_fusion = reads.join(ch_fusion_list.fusions )
+            ch_reads_fusion = reads.join(ch_fusion_list)
         }
 
-        FUSIONINSPECTOR( ch_reads_fusion, index)
+        FUSIONINSPECTOR( ch_reads_fusion, ch_starfusion_ref)
         ch_versions = ch_versions.mix(FUSIONINSPECTOR.out.versions)
 
-        AGAT_CONVERTSPGFF2TSV(FUSIONINSPECTOR.out.out_gtf)
-        ch_versions = ch_versions.mix(AGAT_CONVERTSPGFF2TSV.out.versions)
+        if(!skip_vcf) {
```

Comment:
Feel free to ignore, but I am wondering if it would make sense to move this to the main workflow for logic readability

Reply:
I'll leave that up to you if that's okay, I'm not that familiar with all the different tools so your judgement on this is way better than mine

Reply:
Let's leave it there for the moment and see if it "flows" when we try to make new changes